cinnamon-dynamic-wallpaper@TobiZog/files/cinnamon-dynamic-wallpaper@TobiZog/5.4/src/model/main_view_model.py

```diff
@@ -269,3 +278,43 @@ def set_background_gradient(self):
 		except:
 			self.background_settings['primary-color'] = "#000000"
 			self.background_settings['secondary-color'] = "#000000"
+
+
+	def set_login_image(self):
+		""" Writes a path to file in /tmp/cinnamon_dynamic_wallpaper to display the wallpaper on the login screen
+		"""
+		# New config file content
+		file_content = ""
+
+		# Location of the config file
+		file_location = self.WORKING_DIR + "/slick-greeter.conf"
+
+		if self.cinnamon_prefs.login_image:
+			self.refresh_image()
+			
+			if os.path.isfile("/etc/lightdm/slick-greeter.conf"):
+				# File already exists, make a copy of the config
+				with open("/etc/lightdm/slick-greeter.conf", "r") as conf_file:
+					for line in conf_file.readlines():
+						if not line.startswith("background"):
+							file_content += line
+						elif line.endswith("cinnamon_dynamic_wallpaper/login_image.jpg"):
+							# Skip the configuration. It's already perfect!
+							return
+
+			else:
+				# File doesn't exists	
+				file_content = "[Greeter]\n"
+			
+			file_content += "background=/usr/share/pixmaps/cinnamon_dynamic_wallpaper/login_image.jpg"
+
+			# Create the file
+			with open(file_location, "w") as conf_file:
+				conf_file.write(file_content)
+				conf_file.close()
```

Comment:
Just a comment that you don't need to `close()` if you're using a context manager (`with`). Closing the file handle is done for you automatically.